spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/LettuceConnectionConfiguration.java

```diff
@@ -69,13 +70,29 @@ DefaultClientResources lettuceClientResources() {
 		return DefaultClientResources.create();
 	}
 
+	@Bean
+	@ConditionalOnMissingBean(LettuceClientConfigurationBuilder.class)
+	@ConditionalOnProperty(prefix = "spring.redis.lettuce.pool", value = "enabled", havingValue = "false",
+			matchIfMissing = true)
+	LettuceClientConfigurationBuilder lettuceClientConfigurationBuilder() {
```

Comment:
Rather then exposing `LettuceClientConfigurationBuilder` as a bean, we should rather apply a `LettuceClientConfigurationBuilderCustomizer` to it when the pool library is on the classpath (and that new `enabled` property is not `false`).